File: the-hub-frontend/app/types/theme.ts
export type ThemeWeekday =
  | 'monday'
  | 'tuesday'
  | 'wednesday'
  | 'thursday'
  | 'friday'
  | 'saturday'
  | 'sunday'

export interface Theme {
  theme_id: string
  user_id: string
  name: string
  description: string
  color: string
  icon?: string
  days: ThemeWeekday[]
}

export interface ThemeSuggestion {
  name: string
  description: string
  color: string
  days: ThemeWeekday[]
  tasks: string[]
  reasoning: string
}

export interface ApplyTheme {
  name: string
  description: string
  color: string
  days: ThemeWeekday[]
}

export interface ThemeAssignment {
  task_id: string
  theme_name: string
}

export const WEEKDAYS: ThemeWeekday[] = [
  'monday',
  'tuesday',
  'wednesday',
  'thursday',
  'friday',
  'saturday',
  'sunday'
]

export const WEEKDAY_LABELS: Record<ThemeWeekday, string> = {
  monday: 'Monday',
  tuesday: 'Tuesday',
  wednesday: 'Wednesday',
  thursday: 'Thursday',
  friday: 'Friday',
  saturday: 'Saturday',
  sunday: 'Sunday'
}

File: the-hub-backend/internal/models/theme.go
package models

import (
	"time"

	"github.com/google/uuid"
	"github.com/lib/pq"
	"gorm.io/gorm"
)

// Theme represents a group of tasks shown together on the days assigned to it.
// With the one-theme-per-day rule, a weekday may only belong to a single theme.
type Theme struct {
	ID          uuid.UUID      `json:"theme_id" gorm:"primaryKey;type:uuid;default:gen_random_uuid()"`
	UserID      uuid.UUID      `json:"user_id" gorm:"type:uuid;not null"`
	Name        string         `json:"name" gorm:"not null"`
	Description string         `json:"description"`
	Color       string         `json:"color" gorm:"default:'#3B82F6'"`
	Icon        string         `json:"icon"`
	Days        pq.StringArray `json:"days" gorm:"type:text[]"` // lowercase weekday names: monday..sunday
	User        User           `json:"-" gorm:"foreignKey:UserID"`
	CreatedAt   time.Time      `json:"-"`
	UpdatedAt   time.Time      `json:"-"`
	DeletedAt   gorm.DeletedAt `json:"-" gorm:"index"`
}

// CreateThemeRequest represents the request body for creating a theme.
type CreateThemeRequest struct {
	Name        string   `json:"name" binding:"required" example:"Deep Work"`
	Description string   `json:"description" example:"Focused development and writing work"`
	Color       string   `json:"color" example:"#3B82F6"`
	Icon        string   `json:"icon" example:"i-lucide-briefcase"`
	Days        []string `json:"days" example:"monday,tuesday"`
}

// UpdateThemeRequest represents the request body for updating a theme.
type UpdateThemeRequest struct {
	Name        *string  `json:"name" example:"Deep Work"`
	Description *string  `json:"description"`
	Color       *string  `json:"color"`
	Icon        *string  `json:"icon"`
	Days        []string `json:"days"`
}
